### ISSUE TITLE
Travlr, Scope and Hack The North

### ISSUE THREAD

**Issue by AranScope** [AranScope/aranlong.co.uk#7]
It's currently 12:43pm and we've just finished up with our project and presentations at Hack The North. Needless to say, it was... -- Fast forward a couple of weeks and some much required sleep later.

-- incredible! After my longest stretch of time without sleeping I'm going to share some lessons I learned from 36 hours of ups and downs. The first and in my view, most important, is on scope: It starts with a little story - 

It's 9pm Canadian, we've just arrived after travelling for ~11 hours and stopping over in Toronto. Needless to say, we're very out of it, anyway, we grab some food (mmmm poutine <3) and head straight to the opening ceremony. We spot someone ahead of us gazing in our direction, he walks back and we start chatting. It turns out he had heard our British accents and was curious, well, we ended up working with Chase (he has a name!). The hack started at midnight but we had to sleep so abandoned the first few hours of the hack.

The next day we wandered around the 'job fair' type area, we ended up chatting to pretty much everyone there and decided that the Google cloud tech looked really cool and we'd love to use it. We set to it and had our core architecture designed and working within an hour. That was the end of plain sailing -- as it happens, microservices are a tough sell at a hackathon, each one had its own teething problems required lots of debugging.

In the end, we had an almost finished MVP for an app called 'Travlr', a platform for travellers to get suggestions for day trips, diversions, meet ups and group discounts. All powered by the Google cloud platform, Google maps and Google places with machine learning used to learn the users travelling habits. All of the microservices were built, buy we tragically ran out of time to link these all into our frontend.

Everyone we presented Travlr to -- The HTN judges, Deloitte and a few others, loved the idea. We had a solid presentation from a tech standpoint but fell down in the demos. We now realise that had the idea been minimized to remove the machine learning components and scoped to be reasonably finished within ~16 hours, we could have finished our MVP. 

A fun aside is that the day after we presented Travlr to some members of the Google cloud team and some from the Firebase team, Google Trips was released. Huh, I guess we were onto something after all :)

To summarise, when you're under 24 - 36 hour time constraints, aim to finish to a reasonable standard in half the time. Then you'll have at least a few hours (assuming you overrun as is typical) to debug, round off the sharp corners and work on your presentation.
